src/ucp/wireup/wireup.c

```diff
@@ -1029,6 +1103,103 @@ ucp_wireup_get_reachable_mds(ucp_ep_h ep,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_ep_config_intersect(const ucp_ep_config_key_t *old_key,
+                               const ucp_ep_config_key_t *new_key,
+                               ucp_lane_index_t *intersect_map)
+{
+    ucp_lane_index_t old_lane_idx;
+
+    for (old_lane_idx = 0; old_lane_idx < old_key->num_lanes; ++old_lane_idx) {
+        /* Go through the old configuration and check if the lanes selected for
+         * the old configuration could be used for the new configuration */
+        intersect_map[old_lane_idx] =
+            ucp_wireup_ep_configs_can_reuse_lane(old_key, new_key, old_lane_idx);
+    }
+}
+
+static ucp_lane_index_t
+ucp_wireup_ep_config_find_intersect_lane(const ucp_ep_config_key_t *old_key,
+                                         const ucp_lane_index_t *intersect_map,
+                                         ucp_lane_index_t new_lane)
+{
+    ucp_lane_index_t old_lane_idx;
+
+    for (old_lane_idx = 0; old_lane_idx < old_key->num_lanes; ++old_lane_idx) {
+        if (intersect_map[old_lane_idx] == new_lane) {
+            return old_lane_idx;
+        }
+    }
+
+    return UCP_NULL_LANE;
+}
+
+void ucp_wireup_check_config_intersect(ucp_ep_h ep,
+                                       ucp_ep_config_key_t *new_key)
+{
+    ucp_worker_h worker                            = ep->worker;
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = { UCP_NULL_LANE };
+    ucp_ep_config_key_t *old_key;
+    ucp_lane_index_t lane, reuse_lane;
+    ucp_wireup_ep_t *cm_wireup_ep;
+
+    if (ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) {
+        return;
+    }
+
+    old_key = &ucp_ep_config(ep)->key;
+
+    ucp_wireup_ep_config_intersect(old_key, new_key, reuse_lane_map);
+
+    if (ucp_ep_config_key_has_cm_lane(new_key)) {
```

Comment:
is this func called for non-CM wireup as well?

Reply:
yes

Reply:
but we did not need it before, what's changed?

Reply:
I don't see any problems to check config intersection for non-CM wireup
Also, I think we could support EP reconfiguration now - just need to remove the check for reconfiguration

Reply:
I have no objections, I'm trying to understand side effects of this. WRT ep reconf - it would be great

Reply:
> I'm trying to understand side effects of this

ic, I guess the one side effect is time that is wasted for searching intersections (and trying to re-use lanes)

Reply:
ok then since it's local and number of lanes is limited -

Reply:
> ok then since it's local and number of lanes is limited -

yep

Reply:
if not cm -> skip ucp_wireup_check_config_intersect altogether